Docs: Create new tutorial using foundry + rsk-foundry-starter-kit (#317)
rsksmart/devportal: _quick-start/step1-install-rsk-local-node.md

# show installed versions, and switch to the selected one
# (replace accordingly)
sdk list java | grep installed
sdk use java 8.0.242.j9-adpt
java -version

```

# RSKj

RSKj is the software for running RSK nodes.

There are [several ways](/rsk/node/install/)
to set up a RSK node.
Here, we will download a JAR file,
and run it using the Java SDK that has been installed.

## Download RSKj

Go to the [releases page](https://github.com/rsksmart/rskj/releases),
and click on the most recent one to download it.

You need to click on the JAR file,
in the end of the post about the latest release.
Its name should be `rskj-core-*.jar`:

![Download latest RSK release](/assets/img/tutorials/setup-truffle-oz/image-07.png)

## Verify authenticity of RSKj

When installing and running the RSKj node,
it is always a good idea to verify that your copy is legitimate.

Let's compute the checksum using the following command:

```shell
sha256sum rskj-core-2.0.1-PAPYRUS-all.jar
```

For this version, it looked like this:

```shell
43149abce0a737341a0b063f2016a1e73dae19b8af8f2e54657326ac8eedc8a0 *rskj-core-2.0.1-PAPYRUS-all.jar
```

On Windows, use this command instead:

```windows-command-prompt

> C:>certutil -hashfile C:\RSK\node\rskj-core-2.0.1-PAPYRUS-all.jar SHA256

Reviewer: Reviewers, please test this on Windows.

```

(Replace <PATH-TO-THE-RSKJ-JAR> with your path to the JAR file).

> Example:
>
> ```windows-command-prompt
> C:\>java -cp C:\RSK\node\rskj-core-2.0.1-PAPYRUS-all.jar -Drpc.providers.web.cors=* co.rsk.Start --regtest
>
> ```

If you see no output - that is a good thing:
Its output is directed to a log file.

> Note: The flag provided above, `-Drpc.providers.web.cors=*`,
> This disables cross origin resource sharing protection,
> effectively allowing any web page to access it.
> As we want to make JSON-RPC requests from a browser,
> such as a DApp, we need this flag.

Do **not** close this terminal.
If closed, the RSKj node will stop running.
Run the following commands in a new terminal.

### Connect over HTTP

RSKj allows you to connect over HTTP,
and is currently listening on port `4444`.
Let us verify that this works.
Open a new terminal window.

Issue a JSON-RPC request to the RSKj over HTTP.
For example:

```shell
curl http://localhost:4444/ \
  -s \
  -X POST \
  -H "Content-Type: application/json" \
  --data '{"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}'

```

For Windows:

```windows-command-prompt

> C:>curl http://localhost:4444/ -s -X POST -H "Content-Type: application/json" --data "{\"jsonrpc\":\"2.0\",\"method\":\"eth_blockNumber\",\"params\":[],\"id\":1}"

Reviewer: Reviewers, please test this on Windows as well.

Reviewer 2: It works!
But the correct prompt in Windows is C:\>
You need to add \

Reviewer: gotcha!

```

## cURL

This is a system command that is likely already installed on your system.

If `curl --version` displays an error,
[download curl](https://curl.haxx.se/download.html).

## Install Java on Windows

Go to the official
[Java download page](https://www.java.com/en/download/),
and download and run the installer from there.

![Java Download](/assets/img/tutorials/setup-truffle-oz/image-02.png)

## Install Java on Mac OSX and Linux

There are a variety of ways to do this, and
[SDKman](https://get.sdkman.io/)
is one which allows you to install and
switch between multiple versions as needed:

```shell
curl -s "https://get.sdkman.io/" | bash
source "$HOME/.sdkman/bin/sdkman-init.sh"
# to get a filtered list of available java versions
sdk list java  | grep "8\." # copy a selection for use below

# install the version of java copied above
# (replace accordingly)
sdk install java 8.0.242.j9-adpt

# show installed versions, and switch to the selected one
# (replace accordingly)
sdk list java | grep installed
sdk use java 8.0.242.j9-adpt
java -version

```

# RSKj

RSKj is the software for running RSK nodes.

There are [several ways](/rsk/node/install/)

> to set up a RSK node.

Reviewer 3: ```suggestion
to set up an RSK node.
```

Here, we will download a JAR file,
and run it using the Java SDK that has been installed.

## Download RSKj

Go to the [releases page](https://github.com/rsksmart/rskj/releases),
and click on the most recent one to download it.

You need to click on the JAR file,
in the end of the post about the latest release.
Its name should be `rskj-core-*.jar`:

![Download latest RSK release](/assets/img/tutorials/setup-truffle-oz/image-07.png)

## Verify authenticity of RSKj

When installing and running the RSKj node,

> it is always a good idea to verify that your copy is legitimate.

Reviewer 3: ```suggestion
it is always good practice to verify that your copy is legitimate.
```

```

For Windows:

```windows-command-prompt
C:>curl http://localhost:4444/ -s -X POST -H "Content-Type: application/json" --data "{\"jsonrpc\":\"2.0\",\"method\":\"eth_blockNumber\",\"params\":[],\"id\":1}"

```

The response should look similar to this:

```json
{ "jsonrpc": "2.0", "id": 1, "result": "0x2991b" }
```

The `result` property is the number of the latest block that has been synced. Note that this value (`0x2991b`) is the block number in hexadecimal (base 16), so the output above indicates that the current block number is `170267` in decimal (base 10).

- To get more information about this process:
  Check out
  [set up RSKj with Java](/rsk/node/install/java/).
- If you encounter any problems, check if your system meets the
  [minimum requirements](/rsk/node/install/requirements/).
- There are other ways to install an RSK node,
  on other supported platforms:
  Check out [installing RSKj](/rsk/node/install/).

> Note: You can enter any other valid JSON-RPC commands as the `POST` body.

### Connect over Websockets

RSKj also allows you to connect over websockets,
and is currently listening on port `4445`.

> Let us verify that this works too.

Reviewer 3: ```suggestion
Let's verify that this works too.
```
